File: 1]. DSA/1]. Data Structures/01]. Array/JavaScript/_003)_Delete_Element.js
function deleteElement(arr, element) {
    if (arr.indexOf(element) === -1) {
        return 'no such element exist in array';
    }
    let newArr = [];
    for (let i = 0; i < arr.length; i++) {
        if (i !== arr.indexOf(element)) {
            newArr.push(arr[i]);
        }
    }
    return newArr;
}

let arr = [1, 2, 3, 5];

console.log(arr);
console.log(`before deletion -> ${arr}`);
let deleteElemArr = deleteElement(arr, 25);
console.log(`after deletion -> ${deleteElemArr}`);

File: 1]. DSA/1]. Data Structures/01]. Array/Python/_003)_Delete_Element.py
"""
Function to delete an element in an array

Takes in an array and an element to delete

Returns an new Array
"""


def delete(arr, element):
    if element not in arr:
        return "no such element exist in array"

    new_arr = []

    for index in range(len(arr)):
        if arr[index] != element:
            new_arr.append(arr[index])

    return new_arr


arr = [1, 2, 3, 5]
element = 2

print("Before deletion:", arr)
print("Deleting:", element)

new_arr = delete(arr, element)
print("After deletion:", new_arr)
